lib/pdfkit/middleware.rb

```diff
@@ -19,35 +19,40 @@ def _call(env)
       set_request_to_render_as_pdf(env) if render_as_pdf?
       status, headers, response = @app.call(env)
 
-      if rendering_pdf? && headers['Content-Type'] =~ /text\/html|application\/xhtml\+xml/
-        body = response.respond_to?(:body) ? response.body : response.join
-        body = body.join if body.is_a?(Array)
-
-        root_url = root_url(env)
-        protocol = protocol(env)
-        options = @options.merge(root_url: root_url, protocol: protocol)
-
-        if headers['PDFKit-javascript-delay']
-          options.merge!(javascript_delay: headers.delete('PDFKit-javascript-delay').to_i)
-        end
-
-        body = PDFKit.new(body, options).to_pdf
-        response = [body]
-
-        if headers['PDFKit-save-pdf']
-          File.open(headers['PDFKit-save-pdf'], 'wb') { |file| file.write(body) } rescue nil
-          headers.delete('PDFKit-save-pdf')
-        end
-
-        unless @caching
-          # Do not cache PDFs
-          headers.delete('ETag')
-          headers.delete('Cache-Control')
+      begin
```

Comment:
I don't think this `begin` is necessary. We can just catch exceptions in the whole method.

Also, since we are returning an error, can we put the error in the body?

Can we catch just StandardError as well?

Reply:
Thanks for the review @serene, I've amended as suggested.

Reply:
@serene without `begin` the middleware catches the exceptions on the `@app.call()` above, so it will try to handle exceptions outside of the middleware.

This causes the headers to be nil when exception is triggered, which in turn crashes other middleware down the stack.